alvr/filesystem-layout/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+    pub openvr_driver_dir: std::path::PathBuf,
+    // directory containing default presets
+    pub presets_dir: std::path::PathBuf,
+    // (linux only) path to the executable to wrap vrcompositor
+    pub vrcompositor_wrapper: std::path::PathBuf,
+    // path to the launcher executable
+    pub launcher_exe: std::path::PathBuf,
+    // directory containing dashboard static resources (javascript, css, images)
+    pub dashboard_resources_dir: std::path::PathBuf,
+}
+
+impl Layout {
+    // path to the shared library to be loaded by openVR
+    pub fn openvr_driver_lib(&self) -> std::path::PathBuf {
+        if cfg!(windows) {
+            self.openvr_driver_dir
+                .join("bin/win64/driver_alvr_server.dll")
+        } else {
+            self.openvr_driver_dir
+                .join("bin/linux64/driver_alvr_server.so")
+        }
+    }
+    // path to the manifest file for openVR
+    pub fn openvr_driver_manifest(&self) -> std::path::PathBuf {
+        self.openvr_driver_dir.join("driver.vrdrivermanifest")
+    }
+}
+
+lazy_static! {
+    pub static ref LAYOUT: Layout = if cfg!(windows) {
+        Layout {
+            openvr_driver_dir: std::path::PathBuf::from(""),
+            presets_dir: std::path::PathBuf::from("presets"),
+            vrcompositor_wrapper: std::path::PathBuf::from(""),
+            launcher_exe: std::path::PathBuf::from("ALVR Launcher.exe"),
+            dashboard_resources_dir: std::path::PathBuf::from("share/alvr/dashboard"),
+        }
+    } else {
+        Layout {
+            openvr_driver_dir: std::path::PathBuf::from("lib64/alvr"),
+            presets_dir: std::path::PathBuf::from("share/alvr/presets"),
+            vrcompositor_wrapper: std::path::PathBuf::from("libexec/alvr/vrcompositor-wrapper"),
+            launcher_exe: std::path::PathBuf::from("bin/alvr_launcher"),
+            dashboard_resources_dir: std::path::PathBuf::from("share/alvr/dashboard"),
+        }
+    };
+}
+
+pub fn session_log(alvr_dir: &std::path::Path) -> std::path::PathBuf {
+    if cfg!(windows) {
+        alvr_dir.join("session_log.txt")
+    } else {
+        dirs::home_dir()
+            .expect("get home directory")
+            .join("alvr_session_log.txt")
+    }
+}
+
+pub fn crash_log(alvr_dir: &std::path::Path) -> std::path::PathBuf {
+    if cfg!(windows) {
+        alvr_dir.join("crash_log.txt")
+    } else {
+        dirs::home_dir()
+            .expect("get home directory")
+            .join("alvr_crash_log.txt")
+    }
+}
+
+// return the base alvr path from the path of one component
+pub fn alvr_dir_from_component(
+    component_path: &std::path::PathBuf,
+    component_layout: &std::path::PathBuf,
+) -> Result<std::path::PathBuf, String> {
+    if component_layout.is_absolute() {
+        return Ok(std::path::PathBuf::from("/"));
+    }
+    let mut res = component_path.clone();
+    for component in component_layout.ancestors() {
+        if component == std::path::Path::new("") {
```

Comment:
Use a `use` for this and PathBuf. (rust-analyzer can do this automatically)

Reply:
done